src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -0,0 +1,656 @@
+        "germanynorth": "germanywestcentral",
+        "uaecentral": "uaecentral",
+        "eastus2euap": "eastus2euap",
+        "brazilsoutheast": "brazilsoutheast"
+    }
+
+    # mapping for azure china cloud
+    # log analytics only support China East2 region
+    AzureChinaLocationToOmsRegionCodeMap = {
+        "chinaeast": "EAST2",
+        "chinaeast2": "EAST2",
+        "chinanorth": "EAST2",
+        "chinanorth2": "EAST2"
+    }
+    AzureChinaRegionToOmsRegionMap = {
+        "chinaeast": "chinaeast2",
+        "chinaeast2": "chinaeast2",
+        "chinanorth": "chinaeast2",
+        "chinanorth2": "chinaeast2"
+    }
+
+    # mapping for azure us governmner cloud
+    AzureFairfaxLocationToOmsRegionCodeMap = {
+        "usgovvirginia": "USGV",
+        "usgovarizona": "PHX"
+    }
+    AzureFairfaxRegionToOmsRegionMap = {
+        "usgovvirginia": "usgovvirginia",
+        "usgovtexas": "usgovvirginia",
+        "usgovarizona": "usgovarizona"
+    }
+
+    rg_location = get_rg_location(cmd.cli_ctx, resource_group_name)
+    cloud_name = cmd.cli_ctx.cloud.name
+
+    if cloud_name.lower() == 'azurecloud':
+        workspace_region = AzureCloudRegionToOmsRegionMap.get(
+            rg_location, "eastus")
+        workspace_region_code = AzureCloudLocationToOmsRegionCodeMap.get(
+            workspace_region, "EUS")
+    elif cloud_name.lower() == 'azurechinacloud':
+        workspace_region = AzureChinaRegionToOmsRegionMap.get(
+            rg_location, "chinaeast2")
+        workspace_region_code = AzureChinaLocationToOmsRegionCodeMap.get(
+            workspace_region, "EAST2")
+    elif cloud_name.lower() == 'azureusgovernment':
+        workspace_region = AzureFairfaxRegionToOmsRegionMap.get(
+            rg_location, "usgovvirginia")
+        workspace_region_code = AzureFairfaxLocationToOmsRegionCodeMap.get(
+            workspace_region, "USGV")
+    else:
+        logger.error(
+            "AKS Monitoring addon not supported in cloud : %s", cloud_name)
+
+    default_workspace_resource_group = 'DefaultResourceGroup-' + workspace_region_code
+    default_workspace_name = 'DefaultWorkspace-{0}-{1}'.format(
+        subscription_id, workspace_region_code)
+
+    default_workspace_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.OperationalInsights' \
+                                    '/workspaces/{2}'.format(subscription_id,
+                                                             default_workspace_resource_group, default_workspace_name)
+    resource_groups = cf_resource_groups(cmd.cli_ctx, subscription_id)
+    resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+    # check if default RG exists
+    if resource_groups.check_existence(default_workspace_resource_group):
+        try:
+            resource = resources.get_by_id(
+                default_workspace_resource_id, '2015-11-01-preview')
+            return resource.id
+        except (CloudError, HttpResponseError) as ex:
```

Comment:
We catch both `CloudError` and `HttpResponseError`. I suppose `CloudError` is used in track 1 SDK and `HttpResponseError` is used in track 2.

Here is also the bug fix for the monitoring addon. Since now this may throw `ResourceNotFoundError` (child class of `HttpResponseError`) if the resource group used by log analytics workspace does not exist (just try to get it, if it doesn't exist, will create one next), but we still catch `CloudError` in [current implementation](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/custom.py#L3502) which would result in abnormal termination.

Reply:
> We catch both CloudError and HttpResponseError. I suppose CloudError is used in track 1 SDK and HttpResponseError is used in track 2.

Yes, exactly. But why do we need to add the compatible logic for track 1?

Reply:
Just worried about whether there are other requests would raise `CloudError`.

@zhoxing-ms I'd like to confirm, have all the SDKs used by each module been migrated to track 2?

Reply:
No, which RP SDKs will be called here?

Reply:
For this case, we are calling `ResourceManagementClient`.

Reply:
Removing deprecated `CloudError` in commit [#32797f9](https://github.com/Azure/azure-cli/pull/20433/commits/32797f9645257b58b76080a3cc3fce1887fcada7).